File: server-pedidos-java/src/main/java/br/edu/utfpr/serverpedidosjava/controller/dto/CategoriaDTO.java
package br.edu.utfpr.serverpedidosjava.controller.dto;

import br.edu.utfpr.serverpedidosjava.entity.Categoria;
import jakarta.validation.constraints.NotBlank;

public record CategoriaDTO(
        Integer id,
        @NotBlank(message = "{nome.notblank}") String nome) {

    public Categoria toEntity() {
        Categoria categoria = new Categoria();
        categoria.setId(this.id);
        categoria.setNome(this.nome);
        return categoria;
    }

    public static CategoriaDTO fromEntity(Categoria categoria) {
        return new CategoriaDTO(categoria.getId(), categoria.getNome());
    }
}

File: server-pedidos/src/main/kotlin/br/edu/utfpr/serverpedidos/controller/dto/CategoriaDTO.kt
package br.edu.utfpr.serverpedidos.controller.dto

import br.edu.utfpr.serverpedidos.entity.Categoria
import jakarta.validation.constraints.NotBlank

data class CategoriaDTO(
    val id: Int = 0,
    @field:NotBlank(message = "{nome.notblank}")
    val nome: String = ""
) {
    fun toEntity(): Categoria = Categoria(
        id = this.id,
        nome = this.nome
    )

    companion object {
        fun fromEntity(categoria: Categoria): CategoriaDTO = CategoriaDTO(
            id = categoria.id,
            nome = categoria.nome
        )
    }
}
